### PR TITLE
Adding some impl crumbs to docstrings

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding some impl crumbs to docstrings.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [ ] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/nuclearDataIO/xsCollections.py [hunk 1]
```diff
@@ -381,6 +381,12 @@ def __init__(
     def createMacrosOnBlocklist(
         self, microLibrary, blockList, nucNames=None, libType="micros"
     ):
+        """Create macroscopic cross sections for a list of blocks.
+
+        .. impl:: Build macroscopic cross sections for blocks.
+            :id: I_ARMI_MACRO_XS
+            :implements: R_ARMI_MACRO_XS
```
Comment (albeanth) [inline armi/nuclearDataIO/xsCollections.py:388]: So this requirement looks like it's supposed to live in the macroXSGenerationInterface. Based on that, it looks like this is probably the intended implementation:
https://github.com/terrapower/armi/blob/0c4d5f4c4353f670ce6096b239af360addcc6fde/armi/physics/neutronics/macroXSGenerationInterface.py#L131-L140

This method in xsCollections.py looks relatively unused, actually.

Comment (john-science) [reply armi/nuclearDataIO/xsCollections.py:388]: it is not used in ARMI, but it _is_ used downstream, but some internal projects.

So, there's that. Hmmm

Comment (john-science) [reply armi/nuclearDataIO/xsCollections.py:388]: Actually, that class `MacroXSGenerationInterface` is used  heavily, though only by one downstream project.

Which is interesting, and maybe we could put a not on that  project to see if that should be moved out of ARMI. 

But that file isn't part of this PR.

Comment (john-science) [reply armi/nuclearDataIO/xsCollections.py]: Okay, I moved the `impl` crumb from xsCollections to `MacroXSGenerationInterface.buildMacros())`.

**Review by albeanth — APPROVED** [review]

**Pull request merged**